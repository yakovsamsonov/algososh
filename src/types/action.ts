export enum Action {
  addToHead = 'addToHead',
  addToTail = 'addToTail',
  removeFromHead = 'removeFromHead',
  removeFromTail = 'removeFromTail',
  addToIndex = 'addToIndex',
  removeFromIndex = 'removeFromIndex',
}
