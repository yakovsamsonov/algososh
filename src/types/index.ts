export * from './direction';
export * from './element-states';
export * from './sorting-mode';
export * from './label';
export * from './action';
