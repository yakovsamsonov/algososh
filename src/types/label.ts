export type TLabel = {
  index: number;
  labelType: 'label' | 'circle';
  value: string;
};
