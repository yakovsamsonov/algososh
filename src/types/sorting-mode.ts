export enum SortingMode {
  Selection = 'selection',
  Bubble = 'bubble',
}
