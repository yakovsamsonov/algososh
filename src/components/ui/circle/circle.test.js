import renderer from 'react-test-renderer';

import { Circle } from './circle';
import { ElementStates } from '../../../types';

describe('Круг', () => {
  it('без буквы', () => {
    const circleNoText = renderer.create(<Circle />).toJSON();

    expect(circleNoText).toMatchSnapshot();
  });

  it('с буквой', () => {
    const circleWithText = renderer.create(<Circle letter="345" />).toJSON();

    expect(circleWithText).toMatchSnapshot();
  });

  it('с текстом сверху', () => {
    const circleWithSimpleHead = renderer
      .create(<Circle head={'head'} />)
      .toJSON();

    expect(circleWithSimpleHead).toMatchSnapshot();
  });

  it('с элементом сверху', () => {
    const circleWithElementHead = renderer
      .create(<Circle head={<Circle isSmall letter={'1'}></Circle>} />)
      .toJSON();

    expect(circleWithElementHead).toMatchSnapshot();
  });

  it('с текстом снизу', () => {
    const circleWithSimpleTail = renderer
      .create(<Circle tail={'tail'} />)
      .toJSON();

    expect(circleWithSimpleTail).toMatchSnapshot();
  });

  it('с элементом снизу', () => {
    const circleWithElementTail = renderer
      .create(<Circle tail={<Circle isSmall letter={'2'}></Circle>} />)
      .toJSON();

    expect(circleWithElementTail).toMatchSnapshot();
  });

  it('в индексом', () => {
    const circleWithIndex = renderer.create(<Circle index={0} />).toJSON();

    expect(circleWithIndex).toMatchSnapshot();
  });

  it('маленький', () => {
    const circleSmall = renderer.create(<Circle isSmall />).toJSON();

    expect(circleSmall).toMatchSnapshot();
  });

  it('в базовом состоянии', () => {
    const circleSmall = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();

    expect(circleSmall).toMatchSnapshot();
  });

  it('в состоянии изменения', () => {
    const circleSmall = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();

    expect(circleSmall).toMatchSnapshot();
  });

  it('в состоянии после изменения', () => {
    const circleSmall = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();

    expect(circleSmall).toMatchSnapshot();
  });
});

//в состоянии default;
//в состоянии changing;
//в состоянии modified.
