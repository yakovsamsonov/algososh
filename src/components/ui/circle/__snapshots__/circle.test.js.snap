// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Круг без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг в базовом состоянии 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг в индексом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Круг в состоянии изменения 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг в состоянии после изменения 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг маленький 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с буквой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      345
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с текстом сверху 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с текстом снизу 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Круг с элементом сверху 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          1
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с элементом снизу 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          2
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;
